pkg/controller/networkpolicy/antreanetworkpolicy.go

```diff
@@ -125,20 +127,40 @@ func (n *NetworkPolicyController) deleteANP(old interface{}) {
 // of an UPDATE event.
 func (n *NetworkPolicyController) processAntreaNetworkPolicy(np *crdv1alpha1.NetworkPolicy) *antreatypes.NetworkPolicy {
 	appliedToPerRule := len(np.Spec.AppliedTo) == 0
+	tierPriority := n.getTierPriority(np.Spec.Tier)
 	// appliedToGroupNames tracks all distinct appliedToGroups referred to by the Antrea NetworkPolicy,
 	// either in the spec section or in ingress/egress rules.
 	// The span calculation and stale appliedToGroup cleanup logic would work seamlessly for both cases.
 	appliedToGroupNamesSet := sets.String{}
 	// Create AppliedToGroup for each AppliedTo present in AntreaNetworkPolicy spec.
-	n.processAppliedTo(np.Namespace, np.Spec.AppliedTo, appliedToGroupNamesSet)
+	_, err := n.processAppliedTo(np.Namespace, np.Spec.AppliedTo, appliedToGroupNamesSet)
+	if err != nil {
+		return &antreatypes.NetworkPolicy{
+			SourceRef: &controlplane.NetworkPolicyReference{
+				Type:      controlplane.AntreaNetworkPolicy,
+				Namespace: np.Namespace,
+				Name:      np.Name,
+				UID:       np.UID,
+			},
+			Name:             internalNetworkPolicyKeyFunc(np),
+			UID:              np.UID,
+			Generation:       np.Generation,
+			AppliedToGroups:  appliedToGroupNamesSet.List(),
+			Rules:            make([]controlplane.NetworkPolicyRule, 0, 0),
+			Priority:         &np.Spec.Priority,
+			TierPriority:     &tierPriority,
+			AppliedToPerRule: appliedToPerRule,
+			Realizable:       controlplane.NetworkPolicyEntirelyUnrealizable,
+		}
+	}
 	rules := make([]controlplane.NetworkPolicyRule, 0, len(np.Spec.Ingress)+len(np.Spec.Egress))
 	// Compute NetworkPolicyRule for Ingress Rule.
 	for idx, ingressRule := range np.Spec.Ingress {
 		// Set default action to ALLOW to allow traffic.
 		services, namedPortExists := toAntreaServicesForCRD(ingressRule.Ports)
 		var appliedToGroupNamesForRule []string
 		// Create AppliedToGroup for each AppliedTo present in the ingress rule.
-		atGroups := n.processAppliedTo(np.Namespace, ingressRule.AppliedTo, appliedToGroupNamesSet)
+		atGroups, _ := n.processAppliedTo(np.Namespace, ingressRule.AppliedTo, appliedToGroupNamesSet)
```

Comment:
why do we ignore errors here?

Reply:
Updated with "PartiallyRealizable" status.